private/bufpkg/bufplugin/bufpluginref/bufpluginref.go

```diff
@@ -93,6 +93,18 @@ func PluginReferenceForString(reference string, revision int) (PluginReference,
 	return parsePluginReference(reference, revision)
 }
 
+func PluginReferenceOptionalVersion(rawReference string) (PluginReference, error) {
```

Comment:
Needs godoc

Reply:
Is 0 a valid revision? I think the docs for PluginReference need to be updated, to denote (I think) that the Revision is optional, and will be 0 if unknown, but please verify.